test/integration/apiserver/apiserver_test.go

```diff
@@ -147,7 +147,10 @@ var cascDel = `
 `
 
 func Test4xxStatusCodeInvalidPatch(t *testing.T) {
-	client, _, tearDownFn := setup(t)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Beware, I wasn't precise: NewTestContext does not handle cancellation. That still needs to be handled separately.

Reply:
I would always like to use a withCancel context to make sure derived context will be cancelled in the final. Then it should be ?

```
_, ctx := ktesting.NewTestContext(t)
ctx, cancel := context.WithContext(ctx)
```

Reply:
This will touch a lot of codes, let's make it explicit before the change. By the way, `ktesting.NewTestContext(t)` seems used for tests, could you explain what can we benefit from this?

Reply:
Instead of create a context for each test, I'll return a context from setup. Easy to evolve in the future.